upstairs/src/volume.rs

```diff
@@ -1536,52 +1558,50 @@ impl Volume {
                     )
                 }
 
-                // Presently, we only support one sub_volume for replacement.
-                // If support for multiple sub_volumes is added, then this
-                // following section will need to be updated to loop over the
-                // sub_volume Vec and find the specific one with a difference,
-                // while verifying that all other sub_volumes are no different.
-                if n_sub_volumes.len() != 1 {
-                    crucible_bail!(
-                        ReplaceRequestInvalid,
-                        "Only a single sub_volume is supported"
-                    )
+                let mut sv_results = Vec::new();
+                // Walk the list of SubVolumes, get the compare result from
+                // each one and add it to list for the final result.
+                // If we get an error, then that is returned right away and
+                // we don't bother processing any further.
+                for (o_sv, new_sv) in
+                    o_sub_volumes.iter().zip(n_sub_volumes.iter())
+                {
+                    let sv_res = Self::compare_vcr_for_replacement(
```

Comment:
Any worries about unbounded recursion blowing up the stack here?

Reply:
Yes.  But, I'm not sure what to do about it.
We currently don't have any limits in Nexus, so you could craft VCRs forever until something broke.  I don't know what the limit is currently and what will break first.